### PR TITLE
Fix a bug not to use plugin root directory for buffer path even if configured correctly

### LINKED ISSUE #1408: file output plugin adds a buffer path even if root_dir is enabled
Current Fluentd file buffer plugin may not have `path` in `<buffer>` section, because of `root_dir`.
But file output plugin adds `path` if missing.
It causes unexpected result when users expect to use root dir for buffer paths.

https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/out_file.rb#L103

### PR DESCRIPTION
**Pull request by tagomoris**
fixes #1408.

### CODE REVIEW

**Comment by tagomoris** [comment]
@repeatedly could you review this bug fix?

File: test/plugin/test_out_file.rb [hunk 1]
```diff
@@ -37,6 +37,24 @@ def create_driver(conf = CONFIG)
       assert_equal :gzip, d.instance.instance_eval{ @compress_method }
     end
 
+    test 'using root_dir for buffer path' do
```
Comment (repeatedly) [inline test/plugin/test_out_file.rb]: Does test_out_file have tests for `buffer_conf['path'] = conf['path'] || '/tmp/dummy_path'` line?

Comment (tagomoris) [inline test/plugin/test_out_file.rb]: It's just configuration error for `config_param :path` (in `out_file`), so it's clear that `out_file` raises configuration error for missing path...
But anyway, i added it.

**Comment by tagomoris** [comment]
Can I merge this?

**Comment by repeatedly** [comment]
yes

**Pull request merged**